src/unity/python/turicreate/toolkits/object_detector/util/_visualization.py

```diff
@@ -128,24 +129,33 @@ def draw_bounding_boxes(images, annotations, confidence_threshold=0):
     def draw_single_image(row):
         image = row['image']
         anns = row['annotations']
+        row_number = row['id']
         if anns == None:
             anns = []
         elif type(anns) == dict:
             anns = [anns]
-        pil_img = Image.fromarray(image.pixel_data)
-        _annotate_image(pil_img, anns, confidence_threshold=confidence_threshold)
-        image = _np.array(pil_img)
-        FORMAT_RAW = 2
-        annotated_image = _tc.Image(_image_data=image.tobytes(),
-                                    _width=image.shape[1],
-                                    _height=image.shape[0],
-                                    _channels=image.shape[2],
-                                    _format_enum=FORMAT_RAW,
-                                    _image_data_size=image.size)
+        try:
+            pil_img = Image.fromarray(image.pixel_data)
+            _annotate_image(pil_img, anns, confidence_threshold=confidence_threshold)
+            image = _np.array(pil_img)
+            FORMAT_RAW = 2
+            annotated_image = _tc.Image(_image_data=image.tobytes(),
+                                        _width=image.shape[1],
+                                        _height=image.shape[0],
+                                        _channels=image.shape[2],
+                                        _format_enum=FORMAT_RAW,
+                                        _image_data_size=image.size)
+        except Exception as e:
+            if row_number == -1:
+                # indication that it was a single image and not an SFrame
+                raise _ToolkitError(e)
+            raise _ToolkitError("Received exception at the " + str(row_number) + "th row: " + e)
```

Comment:
I think it's better to say "at row n" than "at the nth row". Namely because it's not always `th`: the `2th row` sounds weird.

Reply:
Fixed!